complex_tensor.py

```diff
@@ -0,0 +1,57 @@
+import torch
+
+
+class ComplexTensor(torch.Tensor):
+    def __new__(cls, re, im):
+        assert (
+            re.device == im.device
+            and re.layout == im.layout
+            and re.requires_grad == im.requires_grad
+            and re.dtype == im.dtype
+        )
+        res = torch.Tensor._make_wrapper_subclass(  # type: ignore[attr-defined]
+            cls,
+            size=re.size(),
+            strides=re.stride(),  # todo: contiguous only
+            storage_offset=0,
+            dtype=torch.complex64,  # todo: real to complex dtype
+            layout=re.layout,
+            device=re.device,
+            requires_grad=False,  # todo: autograd support
```

Comment:
The best way to add autograd support here is to do a parallel to `Tensor` (which is never differentiable.
So I would recommend that `ComplexTensor(...)` is never differentiable and you have a `create_complex_tensor(...)` which is differentiable and built with a custom autograd Function (where you create a ComplexTensor during the fw).